File: VeroApp/Models/Address.swift
import Foundation

struct Address: Codable, Equatable {
    var zipCode: String
    var street: String
    var number: String
    var complement: String
    var neighborhood: String
    var city: String
    var state: String

    var formatted: String {
        var parts = [street]
        if !number.isEmpty      { parts.append(number) }
        if !complement.isEmpty  { parts.append(complement) }
        if !neighborhood.isEmpty { parts.append(neighborhood) }
        if !city.isEmpty || !state.isEmpty { parts.append("\(city) - \(state)") }
        if !zipCode.isEmpty     { parts.append(zipCode) }
        return parts.joined(separator: ", ")
    }
}

File: VeroFlutter/lib/models/address.dart
class Address {
  final String zipCode;
  final String street;
  final String number;
  final String complement;
  final String neighborhood;
  final String city;
  final String state;

  const Address({
    required this.zipCode,
    required this.street,
    required this.number,
    this.complement = '',
    required this.neighborhood,
    required this.city,
    required this.state,
  });

  String get formatted {
    final parts = <String>[
      if (street.isNotEmpty) '$street, $number',
      if (complement.isNotEmpty) complement,
      if (neighborhood.isNotEmpty) neighborhood,
      '$city - $state',
      zipCode,
    ];
    return parts.join(', ');
  }
}
